### DISCUSSION TITLE
AZW3 does not work anymore

### DISCUSSION THREAD

**Discussion by HenningFo** [apprenticeharper/DeDRM_tools#1782] [General]
Hello,
I was greeted by Amazon with a new design of the content and device page. And when I download my content now
DeDRM 7.2.1 does not remove the DRM anymore.
I even tried with an  old eBook which worked before but now doesn't.

**Comment by ElleKayEm** [comment]
Still working for me.  Did you choose an E-ink Kindle (not a Fire) for the device?  Do you have that E-ink Kindle's serial number correctly entered in DeDRM's customization?

**Comment by Anzus** [comment]
Can confirm that on Kindle for Mac it doesn't work anymore, and it looks like the keys are in a new format too.

% ls -a ~/Library/Containers/com.amazon.Kindle/Data/Library/Application\ Support/Kindle/storage
.		..		.kinf2018	.pes2011

They key extraction program is expecting a ".kinf2011" file.

**Reply by j-howell** [reply]
kinf2018 has been supported since DeDRM release 6.8.0 (Jun 18, 2020).

https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.0

**Reply by ElleKayEm** [reply]
Still working for Kindle for Mac too as far as I've heard.  Take a look at this post: https://www.mobileread.com/forums/showpost.php?p=3819708&postcount=508

**Reply by Anzus** [reply]
Hmm, I think I was using it wrong. I went looking to extract the keys manually, and that program is still looking for the kinf2011 file.

**Reply by ElleKayEm** [reply]
Use [kindlekey](https://github.com/apprenticeharper/DeDRM_tools/blob/master/DeDRM_plugin/kindlekey.py) from the 7.2.1 DeDRM plugin.  Or from 6.8.1 if Python 2.

**Reply by Anzus** [reply]
I'll have more time to look at things this evening, but something has changed in Kindle 1.32 for Mac, updated yesterday. For example, Method 2 to prevent KFX no longer works, since renderer-test is missing.

After removing and installing the latest Kindle for Mac from the App store, I get the following:

% ls -l /Applications/Kindle.app/Contents/MacOS
total 46024
-rwxr-xr-x  1 root  wheel    159504 Jul 21 16:25 CrashReporter
-rwxr-xr-x  1 root  wheel  56510880 Jul 21 16:25 Kindle

Thus all downloads are KFX.

**Reply by ElleKayEm** [reply]
I'd stick with the oldest version of Kindle for Mac that you can.

**Reply by ElleKayEm** [reply]
Disabling renderer-test hasn't worked for Kindle for PC for a long time now (1.25?), but other workarounds were found.  I'm still using 1.17 myself for as long as I can.

**Comment by HenningFo** [comment]
It turns out it was a mistake on my side. I recently replaced pycrypto with the forked pycryptodome library. I was not aware of the dependency  relation of DeDRM and pycrypto. After I reinstalled pycrypto alongside pycryptodome DeDRM now again works as before.

-H

**Reply by ElleKayEm** [reply]
Using the scripts on their own?

**Comment by Anzus** [comment]
OK, I uninstalled Calibre and Kindle, re-installed the latest Calibre and the previous version of Kindle, re-installed the DeDRM plugins, and things are working again.

**Comment by lucasbfr** [comment]
Just FYI, the plugin seems to have lost my serial number, updating everything and entering the Serial Number again fixed it!
